Jenkinsfile-nightly-v2

```diff
@@ -0,0 +1,60 @@
+OLTP_TERMINALS = ['1','2','4','8','16']
+
+pipeline {
+    agent none
+
+    options {
+        buildDiscarder(logRotator(daysToKeepStr: '30'))
+        parallelsAlwaysFailFast()
+    }
+
+    triggers {
+        cron('H H(2-3) * * *')
+    }
+
+    stages {
+        stage('Performance') {
+            agent { label 'benchmark' }
+            steps {
+                loop_terminals_run_oltp(OLTP_TERMINALS)
+            }
+            post {
+                cleanup {
+                    deleteDir()
+                }
+            }
+        }
+
+        stage('Microbenchmark') {
+            agent { label 'benchmark' }
+            steps {
+                sh 'echo $NODE_NAME'
+                sh 'echo y | sudo ./script/installation/packages.sh all'
+                sh 'mkdir build'
+                sh 'cd build && cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DTERRIER_USE_ASAN=OFF -DTERRIER_USE_JEMALLOC=ON -DTERRIER_BUILD_TESTS=OFF .. && make -j$(nproc) all'
+                // The micro_bench configuration has to be consistent because we currently check against previous runs with the same config
+                //  # of Threads: 4
+                //  WAL Path: Ramdisk
+                sh 'cd script/micro_bench && ./run_micro_bench.py --run --num-threads=4 --logfile-path=/mnt/ramdisk/benchmark.log'
+                archiveArtifacts 'script/micro_bench/*.json'
+                junit 'script/micro_bench/*.xml'
+            }
+            post {
+                cleanup {
+                    deleteDir()
+                }
+            }
+        }
+    }
+    
+}
+
+def loop_terminals_run_oltp(list){
+    sh 'echo $NODE_NAME'
+    sh 'echo y | sudo ./script/installation/packages.sh all'
+    sh 'mkdir build'
+    sh 'cd build && cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DTERRIER_USE_ASAN=OFF -DTERRIER_USE_JEMALLOC=ON -DTERRIER_BUILD_TESTS=OFF .. && make -j$(nproc) all'
+    sh "cd build && python3 ../script/testing/oltpbench/run_oltpbench_v2.py ../script/testing/oltpbench/nightly_tracefile.json --build-type=release"
```

Comment:
After the meeting today with Andy, maybe we can rename the `--trace-file` to something like `--config-file` since he mentioned that these are more of configs for the tests

Reply:
Moreover, I think we should provide both approaches (with a config file and without one) in the `run_oltpbench_v2.py` to make it more flexible since users may not always have a config on hand when they want to run the oltpbench tests. Beware that if we allow users both approaches, there can be conflicting arguments. What we can do here is to set a priority for the user input in case that a user might supply 2 different values for the same argument, where the input with the higher priority level will override the other one. For example, if a user specifies the `--num-threads` and in a config file and also pass in a different value for the `--num-threads` via the command line optarg, the value from the config file prevails. Meanwhile, if the config file misses an argument but it is supplied via the command line optarg, such user input value will also be effective.

Reply:
Any was pretty explicit about not wanting to support both approaches. I don't have a strong opinion but I assume he is worried that by giving more option of implementations then there will be different uses in different places and it will make it more difficult to maintain (both from the test harness implementation perspective and the use in the Jenkins file perspective)

Reply:
I agree with @bialesdaniel 's points.